src/ol/style/iconstyle.js

```diff
@@ -117,7 +117,7 @@ ol.style.Icon = function(opt_options) {
    * @type {ol.style.IconImage_}
    */
   this.iconImage_ = ol.style.IconImage_.get(
-      image, src, crossOrigin, imageState);
+      image, src, options.imgSize, crossOrigin, imageState);
```

Comment:
And we can add an assertion earlier in the constructor to check that `imgSize` is not defined when `src` is defined.